### ISSUE TITLE
TCP communication error when tcpFastOpen=true

### ISSUE THREAD

**Issue by publicarray** [PowerDNS/pdns#5452]
- Program: dnsdist
 - Issue type: Bug report

### Short description

Set tcpFastOpen=true and query dnsdist using dig with `+tcp` the response is
```
;; communications error to 127.0.0.1#53: end of file
```

### Environment

 - Operating system: 
```sh
[vagrant@dns-resolver ~]$ cat /etc/redhat-release
CentOS Linux release 7.3.1611 (Core) 
[vagrant@dns-resolver ~]$ uname -a
Linux dns-resolver 3.10.0-514.21.1.el7.x86_64 #1 SMP Thu May 25 17:04:51 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
````
 - Software version: 
```
[vagrant@dns-resolver ~]$ dnsdist --version
dnsdist 0.0.1616g6944ee1 (Lua 5.1.4 [LuaJIT 2.0.4])
Enabled features: dnscrypt libsodium protobuf re2 systemd
```
 - Software source:  PowerDNS repository [master branch]

### Steps to reproduce

1. config:
```
setKey("lvQrHBzplDlewnOIwqo0JeVALsAzHhBB3JbfqvOnwWc=")
controlSocket("127.0.0.1")
newServer({address="8.8.8.8", tcpFastOpen=true})
```
2. run `dig google.com @127.0.0.1 +tcp` on the same machine

### Expected behaviour

```
[vagrant@dns-resolver ~]$ dig google.com @8.8.8.8 +tcp

; <<>> DiG 9.9.4-RedHat-9.9.4-38.el7_3.3 <<>> google.com @8.8.8.8 +tcp
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6455
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;google.com.			IN	A

;; ANSWER SECTION:
google.com.		299	IN	A	216.58.196.142

;; Query time: 154 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Fri Jun 23 03:20:33 UTC 2017
;; MSG SIZE  rcvd: 55
```

### Actual behaviour

```
[vagrant@dns-resolver ~]$ dig google.com @127.0.0.1 +tcp -p53
;; communications error to 127.0.0.1#53: end of file
```

### Other information

<details>
<summary>Demo video</summary>

![dnsdist mp4](https://user-images.githubusercontent.com/5497998/27465514-4ef81d3e-5817-11e7-891b-a0e1bbd1c64e.gif)
</details>

**Comment by rgacogne** [comment]
Thanks for reporting this! A fix is available in #5454 and should land in master soon. In the meantime a simple workaround is to not enable `tcpFastOpen` towards backends if `MSG_FASTOPEN` is not available on your distribution.

**Comment by publicarray** [comment]
Thank you for the swift fix 👍
